csrc/multidevice/utils.cpp

```diff
@@ -38,7 +38,9 @@ bool isSharded(TensorView* tv) {
   // allow split/merge if tv is sharded.
   auto tensor_sharded = !is_sharded.empty() && is_sharded.at(0);
   NVF_ERROR(
-      !tensor_sharded || tv->getMaybeRFactorDomain() == tv->getLeafDomain());
+      !tensor_sharded ||
+      (TensorDomain::noReductions(tv->getMaybeRFactorDomain()).at(0) ==
+       TensorDomain::noReductions(tv->getLeafDomain()).at(0)));
```

Comment:
Can you add a code comment? I guess you are trying to make sure all DID transforms are between root and rfactor, but I (and probably other readers) am not sure.

Reply:
overwritten by [Sharded tensor metadata](https://github.com/NVIDIA/Fuser/commit/26c4ca69d584559231467359188915f86c2556ac)

The idea here is to forbid splt/merge on sharded axis